extensions/trace_propagators/src/main/java/io/opentelemetry/extensions/trace/propagation/JaegerPropagator.java

```diff
@@ -189,11 +198,13 @@ private static SpanContext buildSpanContext(String traceId, String spanId, Strin
       int flagsInt = Integer.parseInt(flags);
       TraceFlags traceFlags = ((flagsInt & 1) == 1) ? SAMPLED_FLAGS : NOT_SAMPLED_FLAGS;
 
+      String otelTraceId = StringUtils.padLeft(traceId, MAX_TRACE_ID_LENGTH);
+      String otelSpanId = StringUtils.padLeft(spanId, MAX_SPAN_ID_LENGTH);
+      if (!TraceId.isValid(otelTraceId) || !SpanId.isValid(otelSpanId)) {
+        return SpanContext.getInvalid();
+      }
       return SpanContext.createFromRemoteParent(
```

Comment:
For another PR, we might consider moving the validity checks into `createFromRemoteParent` so propagators don't have to worry about it.

Reply:
That seems reasonable!